staging/src/k8s.io/component-base/metrics/testutil/testutil.go

```diff
@@ -84,3 +86,7 @@ func NewFakeKubeRegistry(ver string) metrics.KubeRegistry {
 
 	return metrics.NewKubeRegistry()
 }
+
+func NewFakePromHttpHandler(registry metrics.KubeRegistry) http.Handler {
```

Comment:
Could we perhaps use the `metrics.Gatherer` interface instead of the KubeRegistry here? That would allow for more flexibility.

Reply:
Yeah. Btw, han shared this has been implemented in https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/component-base/metrics/http.go#L62-L64

We will abandon this request.